ui/src/TestAzureAccount.ts

```diff
@@ -0,0 +1,77 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { SubscriptionClient } from 'azure-arm-resource';
+import { SubscriptionListResult, TenantListResult } from 'azure-arm-resource/lib/subscription/models';
+import { ApplicationTokenCredentials, AzureEnvironment, loginWithServicePrincipalSecret } from 'ms-rest-azure';
+import { Event, EventEmitter } from 'vscode';
+import { AzureAccount, AzureLoginStatus, AzureResourceFilter, AzureSession, AzureSubscription } from './azure-account.api';
+import { ArgumentError } from './errors';
+import { localize } from './localize';
+
+export class TestAzureAccount implements AzureAccount {
+    public status: AzureLoginStatus;
+    public onStatusChanged: Event<AzureLoginStatus>;
+    public waitForLogin: () => Promise<boolean>;
+    public sessions: AzureSession[];
+    public onSessionsChanged: Event<void>;
+    public subscriptions: AzureSubscription[];
+    public onSubscriptionsChanged: Event<void>;
+    public waitForSubscriptions: () => Promise<boolean>;
+    public filters: AzureResourceFilter[];
+    public onFiltersChanged: Event<void>;
+    public waitForFilters: () => Promise<boolean>;
+    private onStatusChangedEmitter: EventEmitter<AzureLoginStatus>;
+    private onFiltersChangedEmitter: EventEmitter<void>;
+
+    public constructor() {
+        this.subscriptions = [];
+        this.status = 'Initializing';
+        this.onStatusChangedEmitter = new EventEmitter<AzureLoginStatus>();
+        this.onStatusChanged = this.onStatusChangedEmitter.event;
+        this.onFiltersChangedEmitter = new EventEmitter<void>();
+        this.onFiltersChanged = this.onFiltersChangedEmitter.event;
+        this.filters = [];
+    }
+
+    public async getTestSubscription(): Promise<void> {
```

Comment:
I think the name is confusing.  How about testLogIn()?

Reply:
I vote for just 'logIn'. The 'test' part feels redundant in the _Test_AzureAccount

I also just get confused by a lot of our `test` naming (my bad, I know), since it feels like `testLogIn` would actually run a test, but this isn't actually testing anything. I'm starting to think we should've named this stuff `MockAzureAccount` and `MockUserInput` or something like that.